src/cplscheme/tests/ExplicitCouplingSchemeTest.cpp

```diff
@@ -355,17 +355,13 @@ BOOST_AUTO_TEST_CASE(testConfiguredSimpleExplicitCoupling)
   std::string configurationPath(_pathToTests + "explicit-coupling-scheme-1.xml");
   std::string nameParticipant0("Participant0");
   std::string nameParticipant1("Participant1");
-  int         dimensions = 3;
 
-  xml::XMLTag          root = xml::getRootTag();
-  PtrDataConfiguration dataConfig(new DataConfiguration(root));
-  dataConfig->setDimensions(dimensions);
-  PtrMeshConfiguration meshConfig(new MeshConfiguration(root, dataConfig));
-  meshConfig->setDimensions(dimensions);
+  xml::XMLTag                                  root = xml::getRootTag();
+  PtrDataConfiguration                         dataConfig(new DataConfiguration(root));
+  PtrMeshConfiguration                         meshConfig(new MeshConfiguration(root, dataConfig));
   m2n::M2NConfiguration::SharedPointer         m2nConfig(new m2n::M2NConfiguration(root));
   precice::config::PtrParticipantConfiguration participantConfig(new precice::config::ParticipantConfiguration(root, meshConfig));
-  participantConfig->setDimensions(dimensions);
```

Comment:
Example 2:

- `DataConfiguration::setDimensions()` does not exist anymore. Data does not have any notion of dimensions, but it takes it from the mesh.
- `MeshConfiguration::setDimensions()` is not needed anymore, as the dimensions are now defined at the mesh tag, which (contrary to the `precice-configuration`/`solver-interface`) *is* available.
- `ParticipantConfiguration::setDimensions()` does not exist anymore. Each participant has multiple meshes, which have their own dimensions.